README.md

```diff
@@ -414,9 +414,11 @@ The `$SERVICE_BINDING_ROOT` environment variable **MUST NOT** be reset if it is
 
 If a `.spec.type` is set, the `type` entry in the workload projection **MUST** be set to its value overriding any existing value.  If a `.spec.provider` is set, the `provider` entry in the workload projection **MUST** be set to its value overriding any existing value.
 
+The `ServiceBinding` status **MUST NOT** be updated as the result of a transient error unless the implementation is unable to retry the binding. A transient error is an api error where retrying may result in success without any other Kubernetes resource being mutated. For example, a conflict when updating the workload resource is transient because retrying may allow the update to succeed, while a not found error is not-transient as the workload will never be found until it is created (which is a resource mutation). An implementation **MAY** record an event as a result of an error.
```

Comment:
I really like this definition:

> **MUST NOT** be updated as the result of a transient error unless the implementation is unable to retry the binding

But I don't necessarily agree with this one:

> transient error is an api error where retrying may result in success without any other Kubernetes resource being mutated

I think categorizing a transient error as being something that doesn't require a k8s resource mutation is too implementation specific - in fact the example of a workload not being found could be a valid transient error where the workload pod was being restarted at the time the binding was created, etc.

Reply:
> the example of a workload not being found could be a valid transient error where the workload pod was being restarted at the time the binding was created, etc. I believe there should be some retry in this case.

There's nothing to retry until the resource is created. This is precisely why it's a non-transient ("permanent") error. An implementation can "watch" for that resource to come into existence and then try again. The bigger point I'm trying to make with this distinction is that when a resource that is expected to exist doesn't, that's an error worth reporting to users for them to take remedial action, while a conflict error can be retried immediately and should not be reported to users.

Reply:
cool - I agree with the idea.  I think perhaps it's the use of the word transient that is throwing me off, because the example of a pod restarting in my mind is a transient error, i.e. it does go away after a while - and I definitely agree that such error should be reported.

Is there another case in your mind besides api errors?  If not, could we say that we don't report api error cases that can be retried?

Reply:
> could we say that we don't report api error cases that can be retried?

All errors could be retired, but the result will be the same until the state of the cluster changes. When user action (or minimally knowledge of an error) is required, that's when we want the condition to be reports on the ServiceBinding's status.

I'm not attached to transient. I've also heard "hard" and "soft" used to describe the type of error. Or "retryable", but that gets a bit circular in definition.

Reply:
> When user action (or minimally knowledge of an error) is required, that's when we want the condition to be reports on the ServiceBinding's status.

I like this definition! 👍    If we can incorporate that into the PR that will be awesome.

Reply:
@arthurdm how does this read?

```suggestion
The `ServiceBinding` status **MUST** be updated for the result of an error when the implementation is unable to retry the binding, or user interaction is required to recover from the error. For example, when updating a workload resource to project a binding a conflict **SHOULD NOT** be reflected because retrying may allow the update to succeed. A not found error fetching the workload resource **SHOULD** be reflected as the error is not recoverable until the target workload resource is created. An implementation **MAY** record an event as a result of any error.
```